lib/react_client.dart

```diff
@@ -564,6 +601,14 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
         component.componentWillUnmount();
       });
 
+  void handleComponentDidCatch(Component2 component, dynamic error, dynamic info) => zone.run(() {
+        component.componentDidCatch(error, info);
+      });
+
+  JsMap handleGetDerivedStateFromError(ComponentStatics2 componentStatics, dynamic error) => zone.run(() {
+        return jsBackingMapOrJsCopy(componentStatics.componentInstance.getDerivedStateFromError(error));
```

Comment:
Why does this one access the `componentInstance` from `componentStatics`, but `handleComponentDidCatch` passes the `component` directly? Seems inconsistent.  Lets pick one unless there is a reason I'm not seeing for why it has to be different.

Reply:
It is different because `getDerivedStateFromError` is static and needs access directly to the instance, whereas the other lifecycle methods don't. The property name wasn't very descriptive to that though, so I updated the name.